ec2api/api/address.py

```diff
@@ -306,6 +306,14 @@ def associate_address(self, context, public_ip=None, instance_id=None,
             for eni in db_api.get_items(context, 'eni'):
                 if instance_id and eni.get('instance_id') == instance_id:
                     instance_network_interfaces.append(eni)
+                    #Bellow changes done for JNT-149.It will restrict user to allocate more than one
+                    # RJIL IP for an instances
+                    eni_id= eni['id']  #store network interface id
+                    for epi in db_api.get_items(context, 'eipalloc'):
+                       #If network interface Id will be present in eipalloc data then it will throw the exception
+                       if 'network_interface_id' in  epi:
+                        if eni_id == epi['network_interface_id']:
```

Comment:
Is this check necessary? In case these two are not equal, we would still throw the same exception, right?